# Texas Public School Discipline Rankings 2017-2018

This report creates rankings for all Texas public schools by discipline types. 
* Rankings are based on the percentage of discipline actions compared to campus cumulative year end enrollment.  
* A school with the lowest rank has the highest percentage of a particular discipline type in the State of Texas. 
* Data used in this report comes from: https://rptsvr1.tea.texas.gov/adhocrpt/Disciplinary_Data_Products/Download_All_Campuses.html
* There were 8,612 campuses considered in this study.
* There were 1200 school districts considered in this study. 

Prepared by:  Jake Drew Ph.D.

In [363]:
#import required Libraries
import pandas as pd
import numpy as np
import os
import string
#Make sure all of our outputs show
pd.set_option('display.max_rows', 500)

## Data Preparation

In [364]:
#Read in raw data file for disipline
path = 'D:/Nathan/HebronSchoolData/RawData/CampusDisciplineStatsAndRanks2018.csv'
discipline2018 = pd.read_csv(path, low_memory=False) 

### This is a Snapshot of the Campus Dataset used for Generating Reports

In [365]:
#Look at the new merged data layout
discipline2018.head()

CAMPUS NAME AND NUMBER  \
0  21ST CENTURY EARLY LEARNING FOUNDA 108913118   
1                          3D ACADEMY 108902006   
2                    6TH GRADE CENTER 116905108   
3                      A & M CONS H S 021901001   
4           A & M CONSOLIDATED MIDDLE 021901042   

   CAMPUS CUMULATIVE YEAR END ENROLLMENT  CAMPUS DISCIPLINE POPULATION  \
0                                  615.0                           NaN   
1                                  262.0                          42.0   
2                                  400.0                          83.0   
3                                 1804.0                         218.0   
4                                  961.0                         140.0   

   CAMPUS DISCIPLINE RECORD COUNT  COUNT OF STUDENTS EXPELLED  \
0                             NaN                         NaN   
1                            92.0                         NaN   
2                           181.0                         NaN   
3                           553.0                         NaN   
4                           514.0                         NaN   

   COUNT OF STUDENTS EXPELLED TO JJAEP  COUNT OF STUDENTS REMOVED TO A DAEP  \
0                                  NaN                                  NaN   
1                                  NaN                                 11.0   
2                                  NaN                                  NaN   
3                                  NaN                                 31.0   
4                                  NaN                                 20.0   

   COUNT OF STUDENTS SUSPENDED IN SCHOOL  DISCRETIONARY DAEP REMOVALS  \
0                                    NaN                          NaN   
1                                   15.0                          NaN   
2                                   66.0                          NaN   
3                                  194.0                          NaN   
4                                  134.0                          NaN   

   DISCRETIONARY EXPULSIONS         ...           JJAEP_Pct_Rank  \
0                       NaN         ...                      NaN   
1                       NaN         ...                      NaN   
2                       NaN         ...                      NaN   
3                       NaN         ...                      NaN   
4                       NaN         ...                      NaN   

   DAEP_Pct_Rank  ISS_Pct_Rank  Discr_DAEP_Pct_Rank  Discr_Expul_Pct_Rank  \
0            NaN           NaN                  NaN                   NaN   
1       0.255716      0.650336                  NaN                   NaN   
2            NaN      0.226115                  NaN                   NaN   
3       0.769555      0.411285                  NaN                   NaN   
4       0.666667      0.298431                  NaN                   NaN   

   Discr_Expul_JJAEP_Pct_Rank Mand_DAEP_Pct_Rank  Mand_Expul_Pct_Rank  \
0                         NaN                NaN                  NaN   
1                         NaN           0.021592                  NaN   
2                         NaN                NaN                  NaN   
3                         NaN                NaN                  NaN   
4                         NaN                NaN                  NaN   

   Mand_Expul_JJAEP_Pct_Rank  Suspension_Pct_Rank  
0                        NaN                  NaN  
1                        NaN             0.136506  
2                        NaN             0.196932  
3                        NaN             0.673763  
4                        NaN             0.392611  

[5 rows x 43 columns]

In [366]:
#Get only the discipline rank fields for review
disciplineRanks2018 = disciplineStats2018[['REGION',
                                           'DISTRICT NAME AND NUMBER',
                                           'CAMPUS NAME AND NUMBER',
                                           'CAMPUS CUMULATIVE YEAR END ENROLLMENT',
                                           'DisciplinePopPct_Rank',
                                           'DisciplineRecordCtPct_Rank',
                                           'ExpelledPct_Rank',
                                           'JJAEP_Pct_Rank',
                                           'DAEP_Pct_Rank',
                                           'ISS_Pct_Rank' ,
                                           'Discr_DAEP_Pct_Rank' ,
                                           'Discr_Expul_Pct_Rank',
                                           'Discr_Expul_JJAEP_Pct_Rank',
                                           'Mand_DAEP_Pct_Rank',
                                           'Mand_Expul_Pct_Rank',
                                           'Mand_Expul_JJAEP_Pct_Rank',
                                           'Suspension_Pct_Rank']]


In [367]:
#Get only the percentage fields for review
disciplinePct2018 = disciplineStats2018[['REGION',
                                         'DISTRICT NAME AND NUMBER',
                                         'CAMPUS NAME AND NUMBER',
                                         'CAMPUS CUMULATIVE YEAR END ENROLLMENT',
                                         'DisciplinePopPct',
                                         'DisciplineRecordCtPct',
                                         'ExpelledPct',
                                         'JJAEP_Pct',
                                         'DAEP_Pct',
                                         'ISS_Pct' ,
                                         'Discr_DAEP_Pct' ,
                                         'Discr_Expul_Pct',
                                         'Discr_Expul_JJAEP_Pct',
                                         'Mand_DAEP_Pct',
                                         'Mand_Expul_Pct',
                                         'Mand_Expul_JJAEP_Pct',
                                         'Suspension_Pct']]

# Hebron High School

## Statewide Discipline Rankings 2017-2018
The following data shows how HHS discipline ranks compared to all other public school campuses in the State of Texas. The first data results show rankings based on percentages of Campus Cumulative Year End Enrollment.  The second data results show discipline categories as a percentage of Campus Cumulative Year End Enrollment. Statewide ranks also include both DAEP and JJAEP campuses.  

**Findings are as follows:**
* Hebron High School ranks in the top 16.6% of all Texas public schools for total campus discipline records.
* Total campus discipline records make up 51% of HHS Campus Cumulative Year End Enrollment.
* Hebron High School ranks in the top 27.7% of all Texas public schools for its total Discipline Population. 
* The Discipline Population makes up 15.7% of HHS Campus Cumulative Year End Enrollment.
* Hebron High School ranks in the top 27.1% of all Texas public schools for its IIS percentage.
* The IIS population makes up 14.8% of HHS Campus Cumulative Year End Enrollment.

In [368]:
disciplineRanks2018[(disciplineRanks2018['CAMPUS NAME AND NUMBER'] == 'HEBRON H S 061902008')] 

REGION DISTRICT NAME AND NUMBER CAMPUS NAME AND NUMBER  \
3306      11    LEWISVILLE ISD 061902   HEBRON H S 061902008   

      CAMPUS CUMULATIVE YEAR END ENROLLMENT  DisciplinePopPct_Rank  \
3306                                 3678.0               0.277219   

      DisciplineRecordCtPct_Rank  ExpelledPct_Rank  JJAEP_Pct_Rank  \
3306                     0.16605               NaN             NaN   

      DAEP_Pct_Rank  ISS_Pct_Rank  Discr_DAEP_Pct_Rank  Discr_Expul_Pct_Rank  \
3306       0.864621      0.271542              0.95528                   NaN   

      Discr_Expul_JJAEP_Pct_Rank  Mand_DAEP_Pct_Rank  Mand_Expul_Pct_Rank  \
3306                         NaN            0.898785                  NaN   

      Mand_Expul_JJAEP_Pct_Rank  Suspension_Pct_Rank  
3306                        NaN             0.832812

In [369]:
disciplinePct2018[(disciplinePct2018['CAMPUS NAME AND NUMBER'] == 'HEBRON H S 061902008')]


REGION DISTRICT NAME AND NUMBER CAMPUS NAME AND NUMBER  \
3306      11    LEWISVILLE ISD 061902   HEBRON H S 061902008   

      CAMPUS CUMULATIVE YEAR END ENROLLMENT  DisciplinePopPct  \
3306                                 3678.0          0.157966   

      DisciplineRecordCtPct  ExpelledPct  JJAEP_Pct  DAEP_Pct   ISS_Pct  \
3306               0.515498          NaN        NaN  0.013594  0.148722   

      Discr_DAEP_Pct  Discr_Expul_Pct  Discr_Expul_JJAEP_Pct  Mand_DAEP_Pct  \
3306        0.004894              NaN                    NaN        0.01006   

      Mand_Expul_Pct  Mand_Expul_JJAEP_Pct  Suspension_Pct  
3306             NaN                   NaN        0.021479

## Region Specific Discipline Rankings 2017-2018
When considering public schools in Region 11, Hebron High School discipline rankings move even higher.

**Findings are as follows:**
* There are 904 public schools considered within Region 11.
* Hebron ranks 95th out of 904 public schools for discipline records as a percentage of Campus Cumulative Year End Enrollment.  
* Hebron High School's total campus discipline records percentage rank falls in the top 13.7% of all Region 11 schools.
* Hebron High School's discipline population percentage rank falls in the top 23.2% of all Region 11 schools.
* Hebron High School's IIS percentage rank falls in the top 22.2% of all Region 11 schools.

In [370]:
#Isolate Region 11
regDisc2018 = pd.DataFrame(disciplinePct2018[(disciplinePct2018['REGION'] == 11)]) 

regDisc2018.reset_index(inplace=True)
regDisc2018.drop(['index'], axis=1, inplace=True)
print('Total schools considered in Region 11: ' + str(len(regDisc2018.index))) 

Total schools considered in Region 11: 904


In [371]:
#Create discipline rankings only within Region 11 
regDisc2018['DisciplinePopPct_Region_Rank'] = \
    regDisc2018['DisciplinePopPct'].rank(ascending=False,pct=True)
                                                   
regDisc2018['DisciplineRecordCtPct_Region_Rank'] = \
    regDisc2018['DisciplineRecordCtPct'].rank(ascending=False,pct=True)

regDisc2018['ISS_Pct_Region_Rank'] = \
    regDisc2018['ISS_Pct'].rank(ascending=False,pct=True)  

    
regDisc2018.drop(['JJAEP_Pct','DAEP_Pct','Discr_DAEP_Pct','ExpelledPct',
                  'Discr_Expul_Pct', 'Discr_Expul_JJAEP_Pct','Mand_DAEP_Pct',
                  'Mand_Expul_Pct','Mand_Expul_JJAEP_Pct', 'Suspension_Pct']
                  , axis=1, inplace=True)

regDisc2018.sort_values(by=['DisciplineRecordCtPct_Region_Rank'], inplace=True)
regDisc2018.reset_index(inplace=True)
regDisc2018.drop(['index'], axis=1, inplace=True)
regDisc2018.head(100)

REGION         DISTRICT NAME AND NUMBER  \
0       11  HURST-EULESS-BEDFORD ISD 220916   
1       11            FORT WORTH ISD 220905   
2       11            FORT WORTH ISD 220905   
3       11             ARLINGTON ISD 220901   
4       11            FORT WORTH ISD 220905   
5       11  HURST-EULESS-BEDFORD ISD 220916   
6       11  HURST-EULESS-BEDFORD ISD 220916   
7       11           CASTLEBERRY ISD 220917   
8       11             ARLINGTON ISD 220901   
9       11            FORT WORTH ISD 220905   
10      11            FORT WORTH ISD 220905   
11      11            LEWISVILLE ISD 061902   
12      11               CROWLEY ISD 220912   
13      11             MANSFIELD ISD 220908   
14      11            FORT WORTH ISD 220905   
15      11           GAINESVILLE ISD 049901   
16      11            FORT WORTH ISD 220905   
17      11              CLEBURNE ISD 126903   
18      11                 VENUS ISD 126908   
19      11             NORTHWEST ISD 061911   
20      11            FORT WORTH ISD 220905   
21      11            FORT WORTH ISD 220905   
22      11               EVERMAN ISD 220904   
23      11            FORT WORTH ISD 220905   
24      11              ALVARADO ISD 126901   
25      11            SPRINGTOWN ISD 184902   
26      11            LEWISVILLE ISD 061902   
27      11               CROWLEY ISD 220912   
28      11            FORT WORTH ISD 220905   
29      11            FORT WORTH ISD 220905   
30      11             BIRDVILLE ISD 220902   
31      11              ALVARADO ISD 126901   
32      11                JOSHUA ISD 126905   
33      11            FORT WORTH ISD 220905   
34      11         MINERAL WELLS ISD 182903   
35      11             MANSFIELD ISD 220908   
36      11            FORT WORTH ISD 220905   
37      11               CROWLEY ISD 220912   
38      11            FORT WORTH ISD 220905   
39      11              GRANBURY ISD 111901   
40      11             ARLINGTON ISD 220901   
41      11             BIRDVILLE ISD 220902   
42      11             BIRDVILLE ISD 220902   
43      11             ARLINGTON ISD 220901   
44      11            FORT WORTH ISD 220905   
45      11             MANSFIELD ISD 220908   
46      11               CROWLEY ISD 220912   
47      11            FORT WORTH ISD 220905   
48      11            FORT WORTH ISD 220905   
49      11            FORT WORTH ISD 220905   
50      11              CLEBURNE ISD 126903   
51      11            LITTLE ELM ISD 061914   
52      11                DENTON ISD 061901   
53      11               CROWLEY ISD 220912   
54      11  HURST-EULESS-BEDFORD ISD 220916   
55      11            FORT WORTH ISD 220905   
56      11             ARLINGTON ISD 220901   
57      11            FORT WORTH ISD 220905   
58      11               CROWLEY ISD 220912   
59      11            LAKE WORTH ISD 220910   
60      11                 KEENE ISD 126906   
61      11            LEWISVILLE ISD 061902   
62      11             MANSFIELD ISD 220908   
63      11             MANSFIELD ISD 220908   
64      11             GLEN ROSE ISD 213901   
65      11            SPRINGTOWN ISD 184902   
66      11             ARLINGTON ISD 220901   
67      11             RIO VISTA ISD 126907   
68      11              CLEBURNE ISD 126903   
69      11           CASTLEBERRY ISD 220917   
70      11             ARLINGTON ISD 220901   
71      11             ARLINGTON ISD 220901   
72      11               EVERMAN ISD 220904   
73      11                 VENUS ISD 126908   
74      11             ARLINGTON ISD 220901   
75      11             ARLINGTON ISD 220901   
76      11                GODLEY ISD 126911   
77      11                JOSHUA ISD 126905   
78      11            CALLISBURG ISD 049905   
79      11         MINERAL WELLS ISD 182903   
80      11             BIRDVILLE ISD 220902   
81      11            LEWISVILLE ISD 061902   
82      11             ARLINGTON ISD 220901   
83      11             MANSFIELD ISD 220908   
84      1

In [372]:
regDisc2018[(regDisc2018['CAMPUS NAME AND NUMBER'] == 'HEBRON H S 061902008')]

REGION DISTRICT NAME AND NUMBER CAMPUS NAME AND NUMBER  \
94      11    LEWISVILLE ISD 061902   HEBRON H S 061902008   

    CAMPUS CUMULATIVE YEAR END ENROLLMENT  DisciplinePopPct  \
94                                 3678.0          0.157966   

    DisciplineRecordCtPct   ISS_Pct  DisciplinePopPct_Region_Rank  \
94               0.515498  0.148722                      0.232061   

    DisciplineRecordCtPct_Region_Rank  ISS_Pct_Region_Rank  
94                           0.137085             0.222028

# Region 11 High School Discipline Rankings 2017-2018
When considering all public high schools in Region 11, Hebron High School discipline rankings improve. 

**Findings are as follows:**

* There are 156 public high schools considered within Region 11.
* Hebron ranks 22nd out of 156 public high schools for discipline records as a percentage of Campus Cumulative Year End Enrollment.  
* Hebron High School's total campus discipline records percentage rank falls in the top 18.1% of all Region 11 high schools.
* Hebron High School's discipline population percentage rank falls in the top 39.4% of all Region 11 schools.
* Hebron High School's IIS percentage rank falls in the top 31% of all Region 11 schools.

In [373]:
#Isolate Region 11 High Schools
regHsDisc2018 = pd.DataFrame(regDisc2018[regDisc2018['CAMPUS NAME AND NUMBER'].str.contains(" H S ", na=False)])

print('Total high schools considered in Region 11: ' + str(len(regHsDisc2018.index))) 

Total high schools considered in Region 11: 156


In [374]:
#Create high school discipline rankings only within Region 11 
regHsDisc2018['DisciplinePopPct_Region_HS_Rank'] = \
    regHsDisc2018['DisciplinePopPct'].rank(ascending=False,pct=True)
                                                   
regHsDisc2018['DisciplineRecordCtPct_Region_HS_Rank'] = \
    regHsDisc2018['DisciplineRecordCtPct'].rank(ascending=False,pct=True)

regHsDisc2018['ISS_Pct_Region_HS_Rank'] = \
    regHsDisc2018['ISS_Pct'].rank(ascending=False,pct=True)
    
regHsDisc2018.drop(['DisciplinePopPct_Region_Rank','DisciplineRecordCtPct_Region_Rank','ISS_Pct_Region_Rank']
                  , axis=1, inplace=True)   

regHsDisc2018.sort_values(by=['DisciplineRecordCtPct_Region_HS_Rank'], inplace=True)
regHsDisc2018.reset_index(inplace=True)
regHsDisc2018.drop(['index'], axis=1, inplace=True)
regHsDisc2018.head(100)

REGION          DISTRICT NAME AND NUMBER  \
0       11               ALVARADO ISD 126901   
1       11             SPRINGTOWN ISD 184902   
2       11             FORT WORTH ISD 220905   
3       11              BIRDVILLE ISD 220902   
4       11                 JOSHUA ISD 126905   
5       11              MANSFIELD ISD 220908   
6       11               GRANBURY ISD 111901   
7       11             FORT WORTH ISD 220905   
8       11             FORT WORTH ISD 220905   
9       11                CROWLEY ISD 220912   
10      11             LEWISVILLE ISD 061902   
11      11              MANSFIELD ISD 220908   
12      11               CLEBURNE ISD 126903   
13      11            CASTLEBERRY ISD 220917   
14      11              ARLINGTON ISD 220901   
15      11                 GODLEY ISD 126911   
16      11              BIRDVILLE ISD 220902   
17      11              ARLINGTON ISD 220901   
18      11              MANSFIELD ISD 220908   
19      11             BRIDGEPORT ISD 249903   
20      11             CALLISBURG ISD 049905   
21      11             LEWISVILLE ISD 061902   
22      11             FORT WORTH ISD 220905   
23      11             FORT WORTH ISD 220905   
24      11              MANSFIELD ISD 220908   
25      11          MINERAL WELLS ISD 182903   
26      11       WHITE SETTLEMENT ISD 220920   
27      11                   KRUM ISD 061905   
28      11                  KEENE ISD 126906   
29      11                 JOSHUA ISD 126905   
30      11              KENNEDALE ISD 220914   
31      11                   AZLE ISD 220915   
32      11             LEWISVILLE ISD 061902   
33      11                  CHICO ISD 249904   
34      11              ARLINGTON ISD 220901   
35      11                CROWLEY ISD 220912   
36      11              BIRDVILLE ISD 220902   
37      11             FORT WORTH ISD 220905   
38      11            PILOT POINT ISD 061903   
39      11              NORTHWEST ISD 061911   
40      11              MANSFIELD ISD 220908   
41      11              NORTHWEST ISD 061911   
42      11            GAINESVILLE ISD 049901   
43      11             FORT WORTH ISD 220905   
44      11                  VENUS ISD 126908   
45      11                EVERMAN ISD 220904   
46      11           STEPHENVILLE ISD 072903   
47      11                 AUBREY ISD 061907   
48      11   HURST-EULESS-BEDFORD ISD 220916   
49      11              ARLINGTON ISD 220901   
50      11                 DUBLIN ISD 072902   
51      11             FORT WORTH ISD 220905   
52      11                   BOYD ISD 249902   
53      11             LAKE WORTH ISD 220910   
54      11                 DENTON ISD 061901   
55      11               GRANBURY ISD 111901   
56      11       EAGLE MT-SAGINAW ISD 220918   
57      11             FORT WORTH ISD 220905   
58      11               BURLESON ISD 126902   
59      11       EAGLE MT-SAGINAW ISD 220918   
60      11              BIRDVILLE ISD 220902   
61      11                 ALVORD ISD 249901   
62      11              ARLINGTON ISD 220901   
63      11              POOLVILLE ISD 184901   
64      11              RIO VISTA ISD 126907   
65      11                 DENTON ISD 061901   
66      11       EAGLE MT-SAGINAW ISD 220918   
67      11             FORT WORTH ISD 220905   
68      11                 DENTON ISD 061901   
69      11                 DENTON ISD 061901   
70      11            WEATHERFORD ISD 184903   
71      11              ARLINGTON ISD 220901   
72      11                 SANGER ISD 061908   
73      11  GRAPEVINE-COLLEYVILLE ISD 220906   
74      11               BURLESON ISD 126902   
75      11  GRAPEVINE-COLLEYVILLE ISD 220906   
76      11              NORTHWEST ISD 061911   
77      11            LAKE DALLAS ISD 061912   
78      11       PREMIER HIGH SCHOOLS 072801   
79      11             LITTLE ELM ISD 061914   
80      11                 JOSHUA ISD 126905   
81      11              MANSFIELD ISD 220908   
82      11          

In [375]:
regHsDisc2018[(regHsDisc2018['CAMPUS NAME AND NUMBER'] == 'HEBRON H S 061902008')]

REGION DISTRICT NAME AND NUMBER CAMPUS NAME AND NUMBER  \
21      11    LEWISVILLE ISD 061902   HEBRON H S 061902008   

    CAMPUS CUMULATIVE YEAR END ENROLLMENT  DisciplinePopPct  \
21                                 3678.0          0.157966   

    DisciplineRecordCtPct   ISS_Pct  DisciplinePopPct_Region_HS_Rank  \
21               0.515498  0.148722                         0.394958   

    DisciplineRecordCtPct_Region_HS_Rank  ISS_Pct_Region_HS_Rank  
21                              0.181818                 0.31068

# Lewisville ISD School Discipline Rankings 2017-2018
When considering all public schools in Lewisville ISD, Hebron High School ranks in the top 9% and is /#2 when compared to other LIDS high schools for discipline records as a percentage of Campus Cumulative Year End Enrollment. 

**Findings are as follows:**
* There are 62 public schools considered within Lewisville ISD.
* Hebron ranks 5th out of 62 public schools for discipline records as a percentage of Campus Cumulative Year End Enrollment.
* Hebron High School's total campus discipline records percentage rank falls in the top 9.4% of all LISD schools.
* Hebron High School's discipline population percentage rank falls in the top 16.6% of all LISD schools.
* Hebron High School's IIS percentage rank falls in the top 17.3% of all LISD schools.
* When considering only LISD high schools, Hebron's discipline records percentage of 51.5% falls at /#2 behind The Colony HS at 63.3%  

In [379]:
#Isolate LewisviLLe ISD High Schools
lewisvillDisc2018 = pd.DataFrame(regDisc2018[regDisc2018['DISTRICT NAME AND NUMBER'] == 'LEWISVILLE ISD 061902'])

print('Total public schools considered in Lewisville ISD: ' + str(len(lewisvillDisc2018.index))) 

Total public schools considered in Lewisville ISD: 62


In [380]:
#Create discipline rankings only within Lewisville ISD 
lewisvillDisc2018['DisciplinePopPct_LISD_Rank'] = \
    lewisvillDisc2018['DisciplinePopPct'].rank(ascending=False,pct=True)
                                                   
lewisvillDisc2018['DisciplineRecordCtPct_LISD_Rank'] = \
    lewisvillDisc2018['DisciplineRecordCtPct'].rank(ascending=False,pct=True)

lewisvillDisc2018['ISS_Pct_LISD_Rank'] = \
    lewisvillDisc2018['ISS_Pct'].rank(ascending=False,pct=True)
    
lewisvillDisc2018.drop(['DisciplinePopPct_Region_Rank','DisciplineRecordCtPct_Region_Rank','ISS_Pct_Region_Rank']
                  , axis=1, inplace=True)   

lewisvillDisc2018.sort_values(by=['DisciplineRecordCtPct_LISD_Rank'], inplace=True)
lewisvillDisc2018.reset_index(inplace=True)
lewisvillDisc2018.drop(['index'], axis=1, inplace=True)
lewisvillDisc2018

REGION DISTRICT NAME AND NUMBER  \
0       11    LEWISVILLE ISD 061902   
1       11    LEWISVILLE ISD 061902   
2       11    LEWISVILLE ISD 061902   
3       11    LEWISVILLE ISD 061902   
4       11    LEWISVILLE ISD 061902   
5       11    LEWISVILLE ISD 061902   
6       11    LEWISVILLE ISD 061902   
7       11    LEWISVILLE ISD 061902   
8       11    LEWISVILLE ISD 061902   
9       11    LEWISVILLE ISD 061902   
10      11    LEWISVILLE ISD 061902   
11      11    LEWISVILLE ISD 061902   
12      11    LEWISVILLE ISD 061902   
13      11    LEWISVILLE ISD 061902   
14      11    LEWISVILLE ISD 061902   
15      11    LEWISVILLE ISD 061902   
16      11    LEWISVILLE ISD 061902   
17      11    LEWISVILLE ISD 061902   
18      11    LEWISVILLE ISD 061902   
19      11    LEWISVILLE ISD 061902   
20      11    LEWISVILLE ISD 061902   
21      11    LEWISVILLE ISD 061902   
22      11    LEWISVILLE ISD 061902   
23      11    LEWISVILLE ISD 061902   
24      11    LEWISVILLE ISD 061902   
25      11    LEWISVILLE ISD 061902   
26      11    LEWISVILLE ISD 061902   
27      11    LEWISVILLE ISD 061902   
28      11    LEWISVILLE ISD 061902   
29      11    LEWISVILLE ISD 061902   
30      11    LEWISVILLE ISD 061902   
31      11    LEWISVILLE ISD 061902   
32      11    LEWISVILLE ISD 061902   
33      11    LEWISVILLE ISD 061902   
34      11    LEWISVILLE ISD 061902   
35      11    LEWISVILLE ISD 061902   
36      11    LEWISVILLE ISD 061902   
37      11    LEWISVILLE ISD 061902   
38      11    LEWISVILLE ISD 061902   
39      11    LEWISVILLE ISD 061902   
40      11    LEWISVILLE ISD 061902   
41      11    LEWISVILLE ISD 061902   
42      11    LEWISVILLE ISD 061902   
43      11    LEWISVILLE ISD 061902   
44      11    LEWISVILLE ISD 061902   
45      11    LEWISVILLE ISD 061902   
46      11    LEWISVILLE ISD 061902   
47      11    LEWISVILLE ISD 061902   
48      11    LEWISVILLE ISD 061902   
49      11    LEWISVILLE ISD 061902   
50      11    LEWISVILLE ISD 061902   
51      11    LEWISVILLE ISD 061902   
52      11    LEWISVILLE ISD 061902   
53      11    LEWISVILLE ISD 061902   
54      11    LEWISVILLE ISD 061902   
55      11    LEWISVILLE ISD 061902   
56      11    LEWISVILLE ISD 061902   
57      11    LEWISVILLE ISD 061902   
58      11    LEWISVILLE ISD 061902   
59      11    LEWISVILLE ISD 061902   
60      11    LEWISVILLE ISD 061902   
61      11    LEWISVILLE ISD 061902   

                          CAMPUS NAME AND NUMBER  \
0                         LEARNING CTR 061902005   
1               MARSHALL DURHAM MIDDLE 061902056   
2                       THE COLONY H S 061902004   
3                      LAKEVIEW MIDDLE 061902046   
4                           HEBRON H S 061902008   
5                       LEWISVILLE H S 061902001   
6                       GRIFFIN MIDDLE 061902044   
7                         DELAY MIDDLE 061902050   
8                      HUFFINES MIDDLE 061902052   
9                   ARBOR CREEK MIDDLE 061902047   
10                        ROCKBROOK EL 061902139   
11                      HEDRICK MIDDLE 061902042   
12                       COLLEGE ST EL 061902102   
13                       LEWISVILLE EL 061902143   
14                 CREEK VALLEY MIDDLE 061902054   
15                  MARJORY VICKERY EL 061902138   
16                          MARCUS H S 061902002   
17                    BRIARHILL MIDDLE 061902049   
18                   FORESTWOOD MIDDLE 061902048   
19                    PETERS COLONY EL 061902109   
20                           POLSER EL 061902125   
21              CLAYTON DOWNING MIDDLE 061902055   
22                      KILLIAN MIDDLE 061902059   
23                     INDIAN CREEK EL 061902113   
24                        LAMAR MIDDLE 061902045   
25                       SOUTHRIDGE EL 061902131   
26                     INDEPENDENCE EL 061902142   
27                         B B OWEN EL 061902115   
28                            DEGAN EL

In [381]:
lewisvillDisc2018[(lewisvillDisc2018['CAMPUS NAME AND NUMBER'] == 'HEBRON H S 061902008')]

REGION DISTRICT NAME AND NUMBER CAMPUS NAME AND NUMBER  \
4      11    LEWISVILLE ISD 061902   HEBRON H S 061902008   

   CAMPUS CUMULATIVE YEAR END ENROLLMENT  DisciplinePopPct  \
4                                 3678.0          0.157966   

   DisciplineRecordCtPct   ISS_Pct  DisciplinePopPct_LISD_Rank  \
4               0.515498  0.148722                    0.166667   

   DisciplineRecordCtPct_LISD_Rank  ISS_Pct_LISD_Rank  
4                          0.09434           0.173913

# Lewisville ISD District Discipline Rankings 2017-2018
When considering all public school districts in Texas, Lewisville ISD ranks in the top 39.9% of districts for discipline records as a percentage of Campus Cumulative Year End Enrollment.

**Findings are as follows:**

* There are 1200 public school districts considered within the State of Texas.
* Lewisville ISD ranks 478th out of 1200 public school districts for discipline records as a percentage of Campus Cumulative Year End Enrollment.
* Lewisville ISD's total campus discipline records percentage rank falls in the top 39.9% of all Texas school districts.
* Lewisville ISD's discipline population percentage rank falls in the top 45.6% of all Texas school districts.
* Lewisville ISD's IIS percentage rank falls in the top 41.4% of all Texas school districts.

In [388]:
#Read in raw data file for district level disipline rankings 
path = 'D:/Nathan/HebronSchoolData/RawData/DistrictDisciplineStatsAndRanks2018.csv'
distDiscipline2018 = pd.read_csv(path, low_memory=False) 
print('Total districts considered in Texas: ' + str(len(distDiscipline2018.index))) 

Total districts considered in Texas: 1200


In [389]:
#Create discipline rankings for Lewisville ISD 
distDiscipline2018 = pd.DataFrame(distDiscipline2018[['REGION',
                                        'DISTRICT NAME AND NUMBER',
                                        'CUMULATIVE YEAR END ENROLLMENT',
                                        'DisciplinePopPct_Rank',
                                        'DisciplineRecordCtPct_Rank',
                                        'ISS_Pct_Rank']])

distDiscipline2018.sort_values(by=['DisciplineRecordCtPct_Rank'], inplace=True)
distDiscipline2018.reset_index(inplace=True)
distDiscipline2018.drop(['index'], axis=1, inplace=True)
distDiscipline2018.head(100)                           

REGION                 DISTRICT NAME AND NUMBER  \
0       16                    BOYS RANCH ISD 180901   
1       12                      CONNALLY ISD 161921   
2        5                      BEAUMONT ISD 123910   
3        4                    TEXAS CITY ISD 084906   
4        4                         DAMON ISD 020910   
5        2                          TAFT ISD 205907   
6       12                       LA VEGA ISD 161906   
7        3                CALHOUN COUNTY ISD 029901   
8        7                    GLADEWATER ISD 092901   
9        3                      BAY CITY ISD 158901   
10      12                WALNUT SPRINGS ISD 018905   
11      13                       WAELDER ISD 089905   
12       5                        JASPER ISD 121904   
13       6                     BUCKHOLTS ISD 166907   
14       4                     GALVESTON ISD 084902   
15      20                        DILLEY ISD 082902   
16       6                      GOODRICH ISD 187903   
17      12                          WACO ISD 161914   
18      14                        SNYDER ISD 208902   
19       7                      MARSHALL ISD 102902   
20       7                       OVERTON ISD 201908   
21      20        BIG SPRINGS CHARTER SCHOOL 193801   
22       6                        HEARNE ISD 198905   
23       8                         PARIS ISD 139909   
24       7                   NACOGDOCHES ISD 174904   
25       6                         SNOOK ISD 026903   
26       3                       WHARTON ISD 241904   
27      10                        ATHENS ISD 107901   
28       7                        WINONA ISD 212910   
29       4                        ALDINE ISD 101902   
30      12                        TEMPLE ISD 014909   
31      13                       BASTROP ISD 011901   
32       6                       TRINITY ISD 228903   
33      10                       KAUFMAN ISD 129903   
34       3                      VICTORIA ISD 235902   
35      15                       COLEMAN ISD 042901   
36      10                   DUNCANVILLE ISD 057907   
37       4                     HITCHCOCK ISD 084908   
38       6                       CALVERT ISD 198902   
39       2                  ARANSAS PASS ISD 205901   
40      13                         ELGIN ISD 011902   
41       8                 LIBERTY-EYLAU ISD 019908   
42       5             WEST ORANGE-COVE CISD 181906   
43       6                   NEW WAVERLY ISD 236901   
44       5                         VIDOR ISD 181907   
45       1                  POINT ISABEL ISD 031909   
46       6                      ROCKDALE ISD 166904   
47      12                       KILLEEN ISD 014906   
48      18                    BIG SPRING ISD 114901   
49      11                         VENUS ISD 126908   
50       7                         RAINS ISD 190903   
51       3                      EL CAMPO ISD 241903   
52       6                         BRYAN ISD 021902   
53       8                   CLARKSVILLE ISD 194904   
54       6          COLDSPRING-OAKHURST CISD 204901   
55       1                         DONNA ISD 108902   
56       7                         TYLER ISD 212905   
57      11                      ALVARADO ISD 126901   
58       7                 SAN AUGUSTINE ISD 203901   
59      13                     DEL VALLE ISD 227910   
60       4                     CLEVELAND ISD 146901   
61       4                     HEMPSTEAD ISD 237902   
62       1                     EDINBURG CISD 108904   
63      10                       RED OAK ISD 070911   
64       7                     PINE TREE ISD 092904   
65       8                MOUNT PLEASANT ISD 225902   
66       2                      WEST OSO ISD 178915   
67      20                   FLORESVILLE ISD 247901   
68       2                    KINGSVILLE ISD 137901   
69      19                      TORNILLO ISD 071908   
70       7                          ALTO ISD 037901   
71      10                        FERRIS ISD

In [390]:
distDiscipline2018[(distDiscipline2018['DISTRICT NAME AND NUMBER'] == 'LEWISVILLE ISD 061902')]

REGION DISTRICT NAME AND NUMBER  CUMULATIVE YEAR END ENROLLMENT  \
478      11    LEWISVILLE ISD 061902                         55092.0   

     DisciplinePopPct_Rank  DisciplineRecordCtPct_Rank  ISS_Pct_Rank  
478               0.456667                    0.399167      0.414167